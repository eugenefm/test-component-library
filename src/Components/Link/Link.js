import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { color, space, typography } from 'styled-system';

const StyledLink = styled('a')(
	{
		textDecoration: 'none'
	},
	color,
	space,
	typography
);

const Link = (props) => {
	const { children, ...others } = props;
	return (
		<StyledLink color='blue' fontFamily='main' {...others}>
			{children}
		</StyledLink>
	);
};

Link.propTypes = {
	children: PropTypes.node.isRequired
};

export default Link;
