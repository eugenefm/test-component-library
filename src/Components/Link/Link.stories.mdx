import { Meta, Story, Props, Preview } from '@storybook/addon-docs/blocks';
import { action } from '@storybook/addon-actions';
import Link from './Link';

<Meta title='Link' component={Link} />

# Link

## Properties

<Props of={Link} />

## Basic usage

<Preview>
	<Story name='Primary'>
		<Link href="https://google.com" target="_blank">
			Call to Action
		</Link>
	</Story>
</Preview>
