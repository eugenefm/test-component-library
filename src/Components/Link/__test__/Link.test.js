import React from 'react';
import ReactDom from 'react-dom';
import { render, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';

import '@testing-library/jest-dom/extend-expect';

import Link from '../Link';

afterEach(cleanup);

it('renders without crashing', () => {
	const div = document.createElement('div');
	ReactDom.render(<Link />, div);
	ReactDom.unmountComponentAtNode(div);
});

it('render Link Component correctly', () => {
	const { getByText } = render(
		<Link href='https://google.com'>Hello World</Link>
	);
	expect(getByText('Hello World')).toExist;
});

it('matches snapshot', () => {
	const tree = renderer
		.create(<Link href='https://google.com'>Call to action</Link>)
		.toJSON();
	expect(tree).toMatchSnapshot();
});
