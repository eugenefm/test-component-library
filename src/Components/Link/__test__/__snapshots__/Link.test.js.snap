// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<a
  className="sc-bdVaJa bxyBnR"
  color="blue"
  fontFamily="main"
  href="https://google.com"
>
  Call to action
</a>
`;
